components/script/dom/bindings/typedarrays.rs

```diff
@@ -103,16 +127,15 @@ impl HeapFloat32Array {
     }
 }
 
-pub fn create_typed_array<T, S>(
+pub fn create_typed_array<T>(
     cx: JSContext,
     data: &[T::Element],
     dest: MutableHandleObject,
-) -> Result<TypedArray<T, S>, ()>
+) -> Result<TypedArray<T, *mut JSObject>, ()>
 where
-    T: TypedArrayElementCreator + TypedArrayElement,
-    S: JSObjectStorage,
+    T: TypedArrayElement + TypedArrayElementCreator,
```

Comment:
Why no leave the generic `JSObjectStorage` bound and use `S` instead of `JSObject`?

Reply:
to avoid adding generic `S` to create_typed_array when I call it from `set_data` or `TextEncoder`